### PR TITLE
Minor improvement for AuditLogEntry#getResponsibleUser

### LINKED ISSUE #1135: AuditLogEntry::getResponsibleUser throw NullPointerException in AuditLogEntryCreateEvent
**To Reproduce:**

1. Listen the event AuditLogEntryCreateEvent
2. try to get the getResponsibleUser 

**Expected Behavior:** if exists get the Optional user or throw NoSuchElementException

**Actual Behavior:** throw NullPointerException

**Version:** 3.2.4

**Other:** Well the issue is mostly because the AuditLogEntry fired in the event not has the AuditLogPart where exists a "cache" of users.

### PR DESCRIPTION
**Pull request by Doc94**
**Description:** This close #1135 by adding docs about when this exception can happen and also handle the null parent for make all return the same type of exception.

### CODE REVIEW

File: core/src/main/java/discord4j/core/object/audit/AuditLogEntry.java [hunk 1]
```diff
@@ -103,8 +103,12 @@ public Optional<Snowflake> getResponsibleUserId() {
      * Gets the user who made the changes, if present.
      *
      * @return The user who made the changes, if present.
+     * @throws NoSuchElementException if the {@link AuditLogEntry#getParent()} not contains the {@link #getResponsibleUserId()} or is null.
      */
     public Optional<User> getResponsibleUser() {
+        if (auditLogPart == null) {
+            throw new NoSuchElementException("The AuditLogPart is not available for this Audit log.");
+        }
```
Comment (quanticc) [inline core/src/main/java/discord4j/core/object/audit/AuditLogEntry.java:111]: What if D4J could return that cached user if `auditLogPart == null`. The entry has a reference to GatewayDiscordClient and could fetch it

What do you think?

Comment (Doc94) [reply core/src/main/java/discord4j/core/object/audit/AuditLogEntry.java:111]: the cached is not a Flux? i dont know how parse this into a Optional.

Comment (quanticc) [reply core/src/main/java/discord4j/core/object/audit/AuditLogEntry.java:111]: Yeah, that's unfortunate. Maybe in a future branch

If this only happens with gateway events, how about we mention this in the exception and javadoc?

Something like: "Audit log entries from gateway events do not include User details, call #getResponsibleUserId instead"

Comment (Doc94) [reply core/src/main/java/discord4j/core/object/audit/AuditLogEntry.java:111]: i mean the throw in javadoc mention this (the @throw) but if is necesary can mention this in better by the NoSuchElementException? i add a minor doc change in header for mention this.

**Review by quanticc — CHANGES_REQUESTED** [review]

**Review by quanticc — APPROVED** [review]

**Pull request merged**